### PR TITLE
added catalogue for installation for oneclick

### PR DESCRIPTION
**Pull request by EraKin575**
## Related Issue
  - Info about Issue or bug

Closes: #1126 

#### Describe the changes you've made
refactored the oneclick installation script to use catalogue

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Please let us know if any test cases are added

Please describe the tests(if any). Provide instructions how its affecting the coverage.

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
A clear and concise description of it.

## Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

## Screenshots (if any)

|        Original         |          Updated           |
|:-----------------------:|:--------------------------:|
| **original screenshot** | <b>updated screenshot </b> |

### CODE REVIEW

File: keploy.sh [hunk 1]
```diff
@@ -1,184 +1,223 @@
...
-            -isCI)
-                IS_CI=true
-                shift
+# Function to display the menu and handle user selection
+platform="Linux"
+OS_NAME="$(uname -s)"
+IS_CI=false
+
+isRunningOnWindows (){
+    if [[ "$OS_NAME" == MINGW32_NT* ]] || [[ "$OS_NAME" == MINGW64_NT* ]]; then
```
Comment (shivamsouravjha) [inline keploy.sh:9]: this isn't just a windows testing rather a OS detection function. Update accordingly

File: keploy.sh [hunk 2]
```diff
@@ -1,184 +1,223 @@
...
+        # Handle CI environment logic here
+        echo "Running in CI environment. Skipping interactive menu."
+        # Additional logic for CI environment
+    fi
+    removefzf
+
+}
 
-installKeploy
+main
```
Comment (shivamsouravjha) [inline keploy.sh:220]: rather than main better to keep it to original name as it justified the usecase

File: keploy.sh [hunk 3]
```diff
@@ -1,184 +1,223 @@
...
+    if [[ "$OS_NAME" == MINGW32_NT* ]] || [[ "$OS_NAME" == MINGW64_NT* ]]; then
+        echo "Windows is not supported. Please use WSL2 instead."
+        exit 1
+    elif [[ "$OS_NAME" == CYGWIN_NT* ]]; then
+        echo "Please install Linux to run this script."
+        exit 1
+    else
+        echo "Detected OS: $OS_NAME"
+    fi
+}
```
Comment (shivamsouravjha) [inline keploy.sh:18]: printing the OS not necessary

File: keploy.sh [hunk 4]
```diff
@@ -1,184 +1,223 @@
...
+OS_NAME="$(uname -s)"
+IS_CI=false
+
+isRunningOnWindows (){
+    if [[ "$OS_NAME" == MINGW32_NT* ]] || [[ "$OS_NAME" == MINGW64_NT* ]]; then
+        echo "Windows is not supported. Please use WSL2 instead."
+        exit 1
+    elif [[ "$OS_NAME" == CYGWIN_NT* ]]; then
+        echo "Please install Linux to run this script."
+        exit 1
```
Comment (shivamsouravjha) [inline keploy.sh:14]: rather than this original code was more correct as this checking is not required.

**Comment by shivamsouravjha** [comment]
Would suggest to refactor to the old logic and just change the display part.

For example keploy won't asked whether to install in docker or in linux incase of mac

with darwin it will only be docker.
image

**Comment by EraKin575** [comment]
@shivamsouravjha quit removed

**Pull request closed without merging**